### ISSUE TITLE
Applicationcommands syncing issues

### ISSUE THREAD

**Issue by NarmyOnDiscord** [Pycord-Development/pycord#1048]
### Summary

After re-starting the bot, sometimes no Applicationcommand is working

### Reproduction Steps

- Add a Slashcommandgroup (I have 2 debug guilds set in my Bot)
- Restart your Bot (may need a few times)
- occasionally, none of the slash commands are working and discord tells you that the application didn't respond

### Minimal Reproducible Code

```python
from discord.ext import commands
import discord

intents = discord.Intents.default()
intents.members = True
intents.typing = False

bot = commands.Bot(command_prefix="!", case_insensitive=True,
                   intents=intents, debug_guilds=[..., ...])  # 2 debug guilds


testGroup = bot.create_group("test", "Description")


@testGroup.command(name="testcmd")
async def test_testcmd(ctx: discord.ApplicationContext):
    await ctx.respond("Hi")

bot.run("Token")
```

### Expected Results

That the command would work and reply with "Hi"

### Actual Results

![image](https://user-images.githubusercontent.com/67862800/154779729-c1e6dbe9-85c8-4bf6-92bd-4f6371926303.png)
Under the red-line is what happened after 2 restarts, couldn't fix it after 2 other restarts

### Intents

discord.Intents.default(), intents.members = True, intents.typing = False

### System Information

- Python v3.10.1-final
- py-cord v2.0.0-beta
    - py-cord pkg_resources: v2.0.0b4
- aiohttp v3.7.4.post0
- system info: Windows 10 10.0.19043

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

**Comment by NarmyOnDiscord** [comment]
> I have this issue or a similiar issue as well, even with await ctx.defer() at the top of the code I still get application failed. Something with pycord from my understanding as my friend who uses node.js doesn't have any issue with application failed.

Yea, a friend of mine who's using another Python Library doesn't have the issue too

**Comment by avenzi** [comment]
I am also experiencing this bug. For me, application commands seem to either register immediately when the bot starts up, or take up to 30 seconds to do so.

**Comment by krittick** [comment]
@BobDotCom Can you confirm if this is being addressed in b5?

**Comment by Revnoplex** [comment]
im getting a simular issue where after several restarts, slash commands break completely and overloads the 200 application command limit. disabling it in a server

**Comment by Ahsoka** [comment]
> It should be fixed in b5, yes

@BobDotCom Did this make it into `v2.0.0b5`?

**Comment by BobDotCom** [comment]
> > It should be fixed in b5, yes
> 
> @BobDotCom Did this make it into `v2.0.0b5`?

I can't reproduce it in b5.

**Comment by BobDotCom** [comment]
> im getting a simular issue where after several restarts, slash commands break completely and overloads the 200 application command limit. disabling it in a server

You could be hitting the 2/60s ratelimit imposed for bulk updates, in which case waiting <60s should fix it.

**Comment by Revnoplex** [comment]
It lasts longer than a few minutes

**Comment by Ahsoka** [comment]
> > > It should be fixed in b5, yes
> > 
> > 
> > @BobDotCom Did this make it into `v2.0.0b5`?
> 
> I can't reproduce it in b5.

Were you able to reproduce it in `v2.0.0b4`?

**Comment by Ahsoka** [comment]
While working on something else I accidently stumbled upon this error whenever this issue happened to me.  This is error appears consistency although it may take several minutes for it to appear after the initial slash command is executed (and ultimately fails).  Although do note that this testing was done in `v2.0.0b4`, however, unless there were specific changes made to `v2.0.0b5` (which I am not aware of) I do not see any reason why this should no longer be an issue `v2.0.0b5`.
```py
ERROR | clovis.automated: [on_error()] The following error occured with the on_interaction event:
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\lib\site-packages\discord\bot.py", line 738, in process_application_commands
    command = self._application_commands[interaction.data["id"]]
KeyError: '955597320072872020'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\lib\site-packages\discord\client.py", line 352, in _run_event
    await coro(*args, **kwargs)
  File "C:\ProgramData\Anaconda3\lib\site-packages\discord\bot.py", line 1048, in on_interaction
    await self.process_application_commands(interaction)
  File "C:\ProgramData\Anaconda3\lib\site-packages\discord\bot.py", line 755, in process_application_commands
    await self.sync_commands(unregister_guilds=[guild_id])
  File "C:\ProgramData\Anaconda3\lib\site-packages\discord\bot.py", line 693, in sync_commands
    await self.http.bulk_upsert_command_permissions(
  File "C:\ProgramData\Anaconda3\lib\site-packages\discord\http.py", line 338, in request
    raise HTTPException(response, data)
discord.errors.HTTPException: 400 Bad Request (error code: 50035): Invalid Form Body
In 0: Invalid application command id provided
```

**Comment by Ahsoka** [comment]
Also for a bit more details on reproduction steps, something that works pretty consistently is the following:
1.  Start up my bot and wait for `on_ready` listener to be triggered.
2. Then end your program and **as fast as possible restart the program**.
3. Then as soon as the `on_ready` listener is triggered try executing a slash command.

If this doesn't work on the first try do not give up.  It may take a couple of tries for the issue to appear.

If I have time I will look into whether or not this happens `v2.0.0b5`, however, my guess is that it should still happen given that no targeted PRs or changes were made to address this issue.

**Comment by Revnoplex** [comment]
> Also for a bit more details on reproduction steps, something that works pretty consistently is the following:
> 
> 1.  Start up your bot and wait for `on_ready` listener to be triggered.
> 
> 2. Then end your program and **as fast as possible restart the program**.
> 
> 3. Then as soon as the `on_ready` listener is triggered try executing a slash command.
> 
> 
> 
> If this doesn't work on the first try do not give up.  It may take a couple of tries for the issue to appear.
> 
> 
> 
> If I have time I will look into whether or not this happens `v2.0.0b5`, however, my guess is that it should still happen given that no targeted >PRs or changes were made to address this issue.
Sounds similar to my issue. Maybe one of the times I restarted the bot just as the on ready occurred just so I could correct something

**Comment by BobDotCom** [comment]
This should be fixed with #1208. Until it is merged (should be soon), you can try to reproduce this using the `fix-registration` branch.
